File: Ai/generativeAi/ui/src/main/java/com/rocks/ui/simplecropview/util/AspectRatio.java
package com.rocks.ui.simplecropview.util;


import androidx.annotation.IntRange;

/**
 * Created by yarolegovich https://github.com/yarolegovich
 * on 06.02.2017.
 */

public class AspectRatio {

    @SuppressWarnings("Range")
    public static final AspectRatio IMG_SRC = new AspectRatio(-1, -1, "Free");

    private final int width;
    private final int height;

    public String getName() {
        return name;
    }
    private final String name;

    public AspectRatio(@IntRange(from = 1) int w, @IntRange(from = 1) int h, String name) {
        this.width = w;
        this.height = h;
        this.name = name;
    }

    public int getWidth() {
        return width;
    }

    public boolean isSquare() {
        return width == height;
    }

    public int getHeight() {
        return height;
    }

    public float getRatio() {
        return ((float) width) / height;
    }
}

File: Ai/generativeAi/data/src/main/java/com/rocks/AspectRatio.kt
package com.rocks

import android.util.Size

class AspectRatio (private val height: Int = 512, private val widthR: Int=1, private val heightR: Int=1){

    private fun getRatio(): Float {

        return widthR.toFloat() / heightR.toFloat()

    }

    private fun getWidth(): Int {
        return (height / getRatio()).toInt()
    }

    fun getSize() = Size(getWidth(),height)

    fun toId() = "$widthR/$heightR"
}